base_layer/core/src/transactions/transaction/transaction_kernel.rs

```diff
@@ -39,13 +40,30 @@ use crate::transactions::{
     transaction_protocol::{build_challenge, TransactionMetadata},
 };
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
+#[repr(u8)]
+pub enum TransactionKernelVersion {
+    DIBBLER = 0,
+}
+
```

Comment:
I think we need a `TransactionKernelVersion::GetCurrentVersion()` implementation here.